src/benchmark/operators/join_benchmark.cpp

```diff
@@ -1,87 +1,84 @@
 #include <memory>
 
-#include "../benchmark_join_fixture.hpp"
 #include "../table_generator.hpp"
 #include "benchmark/benchmark.h"
 #include "operators/join_hash.hpp"
 #include "operators/join_index.hpp"
 #include "operators/join_nested_loop.hpp"
 #include "operators/join_sort_merge.hpp"
 #include "operators/table_wrapper.hpp"
+#include "storage/chunk.hpp"
+#include "storage/index/adaptive_radix_tree/adaptive_radix_tree_index.hpp"
+#include "storage/storage_manager.hpp"
 
 namespace opossum {
 
-BENCHMARK_DEFINE_F(BenchmarkJoinFixture, BM_JoinHash)(benchmark::State& state) {
-  clear_cache();
-
-  auto warm_up =
-      std::make_shared<JoinHash>(_tw_small_uni1, _tw_small_uni2, JoinMode::Inner,
-                                 std::pair<ColumnID, ColumnID>{ColumnID{0}, ColumnID{0}}, PredicateCondition::Equals);
-  warm_up->execute();
-  while (state.KeepRunning()) {
-    auto table_scan =
-        std::make_shared<JoinHash>(_tw_small_uni1, _tw_small_uni2, JoinMode::Inner,
-                                   std::pair<ColumnID, ColumnID>(ColumnID{0}, ColumnID{0}), PredicateCondition::Equals);
-    table_scan->execute();
+void clear_cache() {
+  std::vector<int> clear = std::vector<int>();
+  clear.resize(500 * 1000 * 1000, 42);
+  for (uint i = 0; i < clear.size(); i++) {
+    clear[i] += 1;
```

Comment:
Has this been verified by anyone? Any optimizer worth its salt should throw this away. `_mm_clflush` maybe?

Reply:
I'm pretty sure I validated that the memory really got allocated when I added this function...

Reply:
Maybe allocated, but did it actually flush the cache?

Reply:
No I think I didn't tested that properly. 
From a quick research it appears there is no easy solution and it also depends on the architecture and operating system. Question is if it is worth the effort to be 100% sure the cache is completely flushed?

Reply:
At least we shouldn't pretend to do something that might not happen. We could check if we're on Intel and use `_mm_clflush` there. Not sure about other architectures.

That should happen in a different file and a different PR though (and would fix #700). I just noticed it when reading the PR.